### ISSUE TITLE
Internal Server Error when trying to update certain pages (page won't update)

### ISSUE THREAD

**Issue by thepwnerer** [splitbrain/dokuwiki#3020]
Hey All,

I searched on the forums and I couldn't find anything like this. Everything about the wiki seems to be working fine (I'm on the Greeko b version...I think that's the most up-to-date one). There are certain pages I cannot save at present. It goes to this message:

Internal Server Error
The server encountered an internal error or misconfiguration and was unable to complete your request.

Please contact the server administrator at webmaster@wiki.flfe.net to inform them of the time this error occurred, and the actions you performed just before this error.

More information about this error may be available in the server error log.

Additionally, a 500 Internal Server Error error was encountered while trying to use an ErrorDocument to handle the request.

I've looked at file permissions and it's not that. I've updated PHP to 7.2, so it's not that. It only seems to affect a small number of pages that don't appear to have any discernable differences from the rest of the pages that do work.

Any ideas?

**Comment by splitbrain** [comment]
Thanks for opening the issue. Unfortunately your issue does not contain enough information. Please try to provide as much info as possible, generally following these three main questions:

* What did you do?
* What did you expect to happen?
* What happened instead?

Please also describe what steps you took to analyze the issue and what (failed) attempts you made to fix the problem yourself.

Your issue should contain everything that might help in analyzing and reproducing the issue. That includes but isn't limited to:

* versions of the involved software
* exact error messages
* debug logs (also check your JavaScript console for web related issues)

Thanks for your help.

PS: You may want to refer to [How to Ask Questions the Smart Way](http://www.catb.org/esr/faqs/smart-questions.html) for more tips on improving your issue creation game ;-)

**Comment by phy25** [comment]
> More information about this error may be available in the server error log.

**Comment by thepwnerer** [comment]
I'll try to be as specific as possible:

I was trying to save (after updating) a post that I had created previously. I expected the post to be saved with the content that I adjusted (it won't update). Insted I encountered the internal server error that I looked at above.

I looked in the server log and found this, but it appears to be unrelated:

[Wed Mar 25 11:11:57.815798 2020] [:error] [pid 144147:tid 47584196278016] [client 70.37.110.248:61307] PHP:  syntax error, unexpected '(' in /home/flfe/public_html/.user.ini on line 3

I fixed this by removing the offending character "(" from the .user.ini file and it made no difference.

I looked at permissions in the files that seem to be affected, and they are no different from the permissions in files that are unaffected. So it does not appear to be permissions related. I upgraded my dokuwiki version from greeko a to greeko b. No changes. This is the error in javascript console that comes up:

Failed to load resource: the server responded with a status of 500 ()

I can view the page properly. I can even get into the editor, but when I go to save the editor (whether changes are made or not) then that error page happens above.

Thanks in advance for your help.

Max

**Comment by splitbrain** [comment]
`.user.ini` is not part of DokuWiki but also seems to be unrelated to the 500. So we still don't have the real error message. You did not share the content of the page that causes the trouble. You did not provide any info on the used server and PHP version.

So, we're just guessing here. Could be https://www.dokuwiki.org/faq:mod_security but again, this is just wildly guessing. We need the error message for the 500

**Comment by thepwnerer** [comment]
I am unsure how to find the real error message. This is a shared hosting website, and as far as I know, the only logs I have access to see are the truncated version of the most recent messages from the Errors part of Cpanel. 

Here is the content of the page:

====== How to Record an Email to the Contact Record ======

==== When to ‘Log an Email’ ====
 
  * Emails are an essential part of how we communicate with our customers. This communication needs to have a clear history of the customer’s account so that we can track it.
  * Be clear, concise, informative and easily understandable, keeping in mind that our customers are from all around the world and you are crossing language barriers with your message.
  * Internally, the customers/ accounts you deal with now might not be yours for the entirety of the subscription and therefore, other staff will have to clearly understand your communications.
  * If you are doing something in an account where another staff member is the ‘contact owner’, ensure you ‘NOTIFY’ the other staff member as to what’s been done

==== How to "Log an Email" ====

  * **It should be noted that now your communications are automatically logged, so as long as your Office 365 is synced with Infusionsoft you no longer have to go through this arduous, bug-filled process of using the Infusionsoft for Outlook application. Big hurray.**
  * Once you are in your Microsoft Outlook program and have sent the email to the customer
    * On a side note, you can record as you send an email as well 
  * Go into the ‘Sent’ folder of your email Inbox 
  * It should be the top-most email in the ‘Sent’ folder 
  * Click on it and at the top of your Outlook toolbar, in the middle you will see the “Record Email’ button  
  * Click on this and another window will appear 
  * In the middle of the window, enter in the customer’s first name and last name and search for their account 
  * Once you have found the correct customer, select them from the list by clicking on their name and then the “Use Selected” button 
  * Then F5 (refresh) the account in InfusionSoft to ensure your email was recorded successfully  
  * Scroll down to the bottom of the main page and you will see the ‘Emails’ section at the bottom 
  * The email you just logged should be at the top of the list there now 
  * The next part to this process is to clearly ‘Note’ the customer’s account; Note Type -> Email Note; Note Title -> Email Note/ Subject  
  * See the ‘How-To:  Create a Note’ process  
  * If another staff is the ‘Contact Owner’, they need to be notified of what actions you have taken – ensure that you ‘NOTIFY’ them:  under ‘NOTIFICATIONS’ at the bottom of the window, by selecting the staff member’s name that you want to advise from the drop-down menu  
  * Click ‘Save’  
  * Then F5 (refresh) the account to ensure your changes were accepted successfully

Here is the server information from cpanel:


Hosting Package | 3GB Hosting
-- | --
Server Name | host
cPanel Version | 86.0 (build 16)
Apache Version | 2.4.41
PHP Version | 5.6.40
MySQL Version | 10.3.22-MariaDB
Architecture | x86_64
Operating System | linux
Shared IP Address | 96.30.37.106
Path to Sendmail | /usr/sbin/sendmail
Path to Perl | /usr/bin/perl
Perl Version | 5.16.3
Kernel Version | 3.10.0-1062.9.1.el7.x86_64

PHP version may be misleading. I have upgraded all of our websites to PHP 7.2. I have no ability to upgrade the server's php level.

I appreciate your help.

**Comment by phy25** [comment]
> It only seems to affect a small number of pages that don't appear to have any discernable differences from the rest of the pages that do work.

Could you test by saving the content you posted to pages with a different page name, and save something else to the page that you have problem saving it?

Again without the error message it's very, very hard to guess what is going on. I remembered that in some shared environment, PHP error log might be saved to `error_log` file next to the PHP file that triggered the error, but not sure if that is the case for you.

**Comment by thepwnerer** [comment]
What a great suggestion. I should have thought about that earlier. It seems like I have been able to isolate a particular line which causes the issue. When I put this into a new page I get the same error message:

  * Once you have found the correct customer, select them from the list by clicking on their name and then the 'Use Selected' button 

I haven't had anything show up in that error_log file. I have looked at it.
